api/controllers/CheckUploadedDocumentsController.js

```diff
@@ -14,20 +17,30 @@ const CheckUploadedDocumentsController = {
             if (uploadedFileData.length === 0) {
                 throw new Error('No files uploaded');
             }
-            uploadedFileData.forEach(async (uploadedFile) => {
-                await UploadedDocumentUrls.create(
+            uploadedFileData.forEach((uploadedFile) => {
+                UploadedDocumentUrls.create(
                     CheckUploadedDocumentsController._dbColumnData(
                         uploadedFile,
                         req
                     )
-                );
-                sails.log.info(
-                    `Url for document ${uploadedFile.filename} added to db`
-                );
+                )
+                    .then(() => {
+                        sails.log.info(
+                            `Url for document ${uploadedFile.filename} added to db`
+                        );
+                    })
+                    .catch((err) => {
+                        sails.log.error(err);
+                        res.serverError();
+                    });
             });
-            res.redirect('/check-uploaded-documents');
+
+            CheckUploadedDocumentsController._checkDocumentCountAndPaymentDetails(
+                req,
+                res
+            );
         } catch (err) {
-            sails.log.error(err);
+            sails.log.error(err.message, err.stack);
```

Comment:
Hmmm, I was under the assumption when you throw an Error in a try/catch without a 'finally' it will break the loop by default. Is this not the case 😱

Reply:
An error will break the loop, but this promise won't throw an error synchronously so all files will be processed at the same time and an error in the promise will be unhandled. (I've read that back and it's confusing so if you don't get it we can chat through)

Reply:
Ah I see, yes that does make sense. Okay, I'll look through the code and try and resolve it.